### PR TITLE
KAFKA-16666: Migrate OffsetMessageFormatter to tools module

### PR DESCRIPTION
**Pull request by m1a2st**
https://issues.apache.org/jira/browse/KAFKA-16666

transfer OffsetMessageFormatter form scala code to java code, and move to tool module

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/tools/consumer/OffsetsMessageFormatter.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    /**
+     * We ignore the timestamp of the message because GroupMetadataMessage has its own timestamp.
+     */
+    private Optional<OffsetCommitValue> readToOffsetCommitValue(ByteBuffer byteBuffer) {
+        short version = byteBuffer.getShort();
+        if (version >= OffsetCommitValue.LOWEST_SUPPORTED_VERSION
+                && version <= OffsetCommitValue.HIGHEST_SUPPORTED_VERSION) {
+            OffsetCommitValue value = new OffsetCommitValue(new ByteBufferAccessor(byteBuffer), version);
+            value.setLeaderEpoch(value.leaderEpoch() == NO_PARTITION_LEADER_EPOCH ? 0 : value.leaderEpoch());
```
Comment (chia7712) [inline tools/src/main/java/org/apache/kafka/tools/consumer/OffsetsMessageFormatter.java]: IMHO, we should display the "raw" data. That is good hint in debugging.

File: tools/src/test/java/org/apache/kafka/tools/consumer/OffsetMessageFormatterTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import java.io.PrintStream;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyMap;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class OffsetMessageFormatterTest {
+
+    private static final OffsetCommitKey OFFSET_COMMIT_KEY = new OffsetCommitKey()
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/consumer/OffsetMessageFormatterTest.java:44]: Could you please add test to make sure `OffsetMessageFormatter` can skip `GroupMetadataKey`?

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for this nice patch!

**Comment by m1a2st** [comment]
@chia7712, Thanks for your comments, PTAL

**Pull request merged**